lxd/instance/drivers/driver_qemu.go

```diff
@@ -2721,6 +2820,26 @@ func (d *qemu) generateQemuConfigFile(mountInfo *storagePools.MountInfo, busName
 
 	cfg = append(cfg, qemuDriveConfig(&driveConfig9pOpts)...)
 
+	// If user has requested AMD SEV, check if supported and add to QEMU config.
+	sevOpts := &qemuSevOpts{}
```

Comment:
I can't see why sevOpts is defined outside of the if block below, we should try and limit the scope of variables wherever possible to avoid confusion. Also why don't we modify d.setupSev to return the `qemuSevOpts` struct, rather than pass a pointer to it to be modified? Then you can directly assign it to `sevOpts` here. What do you think?

Reply:
Looks like static analysis has picked up some issues here as well.

Reply:
thanks ! Taking care of all that